pkg/fmap/fmap.go

```diff
@@ -86,6 +86,22 @@ type Metadata struct {
 	Start uint64
 }
 
+// HeaderValid A basic check whether the fmap header is valid
+func HeaderValid(H *Header) bool {
+	const HeaderSize = 56
+	const AreaSize = 42
+
+	if H.VerMajor != 1 {
+		return false
+	}
+	if H.Size < uint32(HeaderSize+H.NAreas*AreaSize) {
```

Comment:
Is there a reason this is `<` rather than `==`?

Reply:
So this is checking if the total flash size is smaller than the size of the fmap header struct + fmap entries struct. '< 'is correct since you generally want something else than an fmap in the flash, but this check might as well have been H.Size > 0...